2016/proposals/verma-nikhil.md

```diff
@@ -0,0 +1,232 @@
+# Manage Workflow for Instructor Training
+
+## Abstract
+
+Software Carpentry has been training scientists with computing skills since the last 18 years, and the number of instructors has been growing rapidly since its inception. More and more people are applying for instructor training programme which makes it difficult to manage the whole process manually. To help ease the problem, this project helps admins manage the whole workflow of instructor training starting from application to final checkout. This would help orgainizing the currently scattered workflow, getting rid of scattered emails and what not.
+
+## Proposed Workflow
+
+The whole system to manage instructor training is supposed to have two interfaces:
+
+1. The Trainee Admin - An administrative interface - which is used by admins to manage trainees at each stage, and
+
+2. The Trainee Dashboard - the one used by trainees to keep track of their progress.
+
+The process starts with applicants applying for instructor training programmes
+
+### 1. The Application Form
+
+The application process for becoming an instructor starts here. People aspiring to become instructors shall start by filling up this application form.
+
+Trainees apply in one of the following ways:
+
+  1. **Apply individually**:
+
+  This would be a normal application, requesting the relevant details from the applicant.
+
+  **Helping individuals form groups**
+
+   * Apart from the regular details, the applicant would also be asked if they are willing to pair up.
+   * If yes, they would be presented with a list of individuals or groups willing to accept new members. The applicant can then request to pair up with the individual/group of their choice.
+
+  **Determine alredy associated applicants**
+
+  The applicant will be presented with a list of partner organizations which will be used to determine whether they are already associated with a partner organization
+
+  2. **Apply in a group**
+
+  Applicants can also apply in groups, making it easier to manage a bunch of people together. This would have the following fields:
+
+   * Individual details of every member, as in [1] above
+   * Group Leader
+   * Whether or not they are willing to expand the group
+
+
+### 2. Managing Trainees - The Trainee Admin
+
+  The Trainee Admin will be used to manage trainees at each stage of training. Access will be based on permissions (to be discussed during GSoC), so that the right people get to change and monitor the trainee status at each stage. For now, an *admin* is the relevant person handling a particular stage for an applicant. The admin workflow is outlined here:
```

Comment:
I agree permissions will be discussed during the project, but it would be helpful to add a couple of points here about what you think they should be.